### PR TITLE
Reenable comment count checks

### PR DESCRIPTION
**Pull request by mmitche**
Need to keep track of the comment count from the GHPullRequest in addition to the comment count coming from the webhook, to avoid missing the first comment.

@DavidTanner The reason that we missed comments before was that we didn't pull the full comment data all the time.  We do this when we get PR data (update/close/open etc.) but not on comment hooks.  This means that the first comment could be missed.

I think this solves the overall problem, though I'm not positive about the non-webhook side of things.

### CODE REVIEW

**Comment by benpatterson** [comment]
@mmitche I'm curious what the comment counts are used for? Still grok'ing over here..thanks for your patience.

**Comment by mmitche** [comment]
@benpatterson This avoids us doing an API call to get the GH comments if there are none.  The number of comments reported by GH is accurate, but the number in the GH object we have may be out of date with respect to the actual number of comments in the PR because we don't get fresh pull request data on every GH webhook event (if we did we would quickly run out of API calls/hr).

I can't remember the exact scenario, but it basically meant we could miss the first comment in some corner cases.

**Comment by mmitche** [comment]
@benpatterson Updated against HEAD.  Depending on which order PRs are merged, this might need another round of changes to the tests because new tests and methods were added in other PRs.

File: src/main/java/org/jenkinsci/plugins/ghprb/GhprbPullRequest.java [hunk 1]
```diff
@@ -259,14 +262,19 @@ private void updatePR(GHPullRequest ghpr, GHIssueComment comment, boolean isWebh
 
             synchronized (this) {
                 boolean wasUpdated = setUpdated(updatedDate);
-
+
```
Comment (samrocketman) [inline src/main/java/org/jenkinsci/plugins/ghprb/GhprbPullRequest.java]: Trailing spaces.  I'm only going to comment this once.  But anywhere there's trailing spaces in this PR should be removed.

File: src/main/java/org/jenkinsci/plugins/ghprb/GhprbPullRequest.java [hunk 2]
```diff
@@ -78,6 +78,9 @@
 
     private final int id;
     private Date updated; // Needed to track when the PR was updated
+    // Local representation of the number of comments that we have
+    // on the PR.
+    private transient int commentCount;
```
Comment (samrocketman) [inline src/main/java/org/jenkinsci/plugins/ghprb/GhprbPullRequest.java]: Note to other maintainers.  In Jenkins plugins, `transient` means the `commentCount` value _will not_ be serialized to disk as XML (i.e. it will not be saved to XML configuration).  This value will not persist across Jenkins master restarts.

**Comment by samrocketman** [comment]
Also, please rebase on `origin/master` so there's no merge conflicts.
